sunxspex/emission.py

```diff
@@ -0,0 +1,814 @@
+    Parameters
+    ----------
+    photon_energies : `numpy.array`
+        Array of photon energies to evaluate flux at
+    p : `float`
+        Slope below the break energy
+    eebrk : `float`
+        Break energy
+    q : `float`
+        Slope above the break energy
+    eelow : `float`
+        Low energy electron cut off
+    eehigh : `float`
+        High energy electron cut off
+    efd : `bool`
+        True (default) - input electron distribution is electron flux density distribution
+        (unit electrons cm^-2 s^-1 keV^-1),
+        False - input electron distribution is electron density distribution.
+        (unit electrons cm^-3 keV^-1),
+        This input is not used in the main routine, but is passed to brm2_dmlin and Brm2_Fthin
+
+    Returns
+    -------
+    flux: `numpy.array`
+        Multiplying the output of Brm2_ThinTarget by a0 gives an array of
+        photon fluxes in photons s^-1 keV^-1 cm^-2, corresponding to the photon energies in the
+        input array eph. The detector is assumed to be 1 AU rom the source. The coefficient a0 is
+        calculated as a0 = nth * V * nnth, where nth: plasma density; cm^-3) V:
+        volume of source; cm^3) nnth: Integrated nonthermal electron flux density (cm^-2 s^-1), if
+        efd = True, or Integrated electron number density (cm^-3), if efd = False
+
+    Notes
+    -----
+    If you want to plot the derivative of the flux, or the spectral index of the photon spectrum as
+    a function of photon energy, you should set RERR to 1.d-6, because it is more sensitive to RERR
+    than the flux.
+
+    Adapted from SSW `Brm2_ThinTarget
+    <https://hesperia.gsfc.nasa.gov/ssw/packages/xray/idl/brm2/brm2_thintarget.pro>`_
+    """
+    mc2 = const.get_constant('mc2')
+    clight = const.get_constant('clight')
+    au = const.get_constant('au')
+
+    # Max number of points
+    maxfcn = 2048
+
+    # Average atomic number
+    z = 1.2
+
+    # Relative error
+    rerr = 1e-4
+
+    # Numerical coefficient for photo flux
+    fcoeff = (clight / (4 * np.pi * au ** 2)) / mc2 ** 2.
+
+    # Create arrays for the photon flux and error flags.
+    flux = np.zeros_like(photon_energies, dtype=np.float64)
+    iergq = np.zeros_like(photon_energies, dtype=np.float64)
+
+    if eelow >= eehigh:
+        raise ValueError('eehigh must be larger than eelow!')
+
+    l, = np.where((photon_energies < eehigh) & (photon_energies > 0))
+    if l.size > 0:
+        flux[l], iergq[l] = split_and_integrate(model='thin-target',
+                                                photon_energies=photon_energies[l], maxfcn=maxfcn,
+                                                rerr=rerr, eelow=eelow, eebrk=eebrk, eehigh=eehigh,
+                                                p=p, q=q, z=z, efd=efd)
+
+        flux *= fcoeff
+
+        return flux
+    else:
+        raise Warning('The photon energies are higher than the highest electron energy or not '
+                      'greater than zero')
```

Comment:
Mabye the if only for the raise and the rest outside the if to remove indentation?
```python
if not l:
    raise Warning...
flux[l], ....
```